conda_recipe_manager/parser/_types.py

```diff
@@ -116,7 +116,7 @@ class Regex:
     """
 
     # Pattern to detect Jinja variable names and functions
-    _JINJA_VAR_FUNCTION_PATTERN: Final[str] = r"[a-zA-Z_][a-zA-Z0-9_\|\'\"\(\)\[\]\, =\.\-]*"
+    _JINJA_VAR_FUNCTION_PATTERN: Final[str] = r"[a-zA-Z_][a-zA-Z0-9_\|\'\"\(\)\[\]\, =\.\-~]*"
```

Comment:
I actually don't see `~` in the proposed CEP-17, so I'm not sure if it will be supported.
I also need to figure out what the hell that actually does in JINJA, lol.

Reply:
Okay so from context of the original post, I think the `~` is tied to semantic versioning. Now why that's valid outside of the string argument in the `pin_subpackage()` example, I'm less sure.